airtime_mvc/application/controllers/BillingController.php

```diff
@@ -19,6 +20,24 @@ public function indexAction()
         $this->_redirect('billing/upgrade');
     }
 
+    public function promoEligibilityCheckAction()
+    {
+        $this->view->layout()->disableLayout();
+        $this->_helper->viewRenderer->setNoRender(true);
+
+        $request = $this->getRequest();
+        if (!$request->isPost()) {
+            throw new Exception("Must POST data to promoEligibilityCheckAction.");
+        }
```

Comment:
You should check for the CSRF token here. 

Check out uploadAction() in PluploadController.php for an example...